test/requests/code_lens_expectations_test.rb

```diff
@@ -127,8 +127,10 @@ def name
       end
 
       def create_code_lens_listener(uri, emitter, message_queue)
+        raise "uri can't be nil" unless uri
```

Comment:
Let's leverage Sorbet for this instead. We can just add a signature and we shouldn't need this check.

Reply:
I've tried it but maybe I was doing something wrong because it's like wrestling with Sorbet with little benefit due to the dynamic class definition:

1. We need to add `extend T::Sig` to `CodeLensExpectationsTest` instead of the extension test class.
2. In this case, we need to add `override` to the signature of `create_code_lens_listener`. But Sorbet would complain that there's nothing to override about, which I think is also due to dynamic class definition?
    ![Screenshot 2023-09-07 at 12 20 23](https://github.com/Shopify/ruby-lsp/assets/5079556/97a5984b-d72f-444e-89ec-b0510de2b470)
3. If there's any runtime type checking error, it will be swallowed in `RubyLsp::Result` instead of being raised

    ![Screenshot 2023-09-07 at 12 21 31](https://github.com/Shopify/ruby-lsp/assets/5079556/07516d66-3f6b-4fdb-b80c-ebf7a3d798f9)

So eventually I decided that it's not worth the effort here.